src/main/java/aztech/modern_industrialization/compat/kubejs/machine/ExtraMachineConfig.java

```diff
@@ -0,0 +1,76 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package aztech.modern_industrialization.compat.kubejs.machine;
+
+import aztech.modern_industrialization.machines.components.CustomOverclockComponent;
+import aztech.modern_industrialization.machines.components.GunpowderOverclockComponent;
+import aztech.modern_industrialization.machines.components.IOverclockComponent;
+import aztech.modern_industrialization.machines.init.MultiblockMachines;
+import aztech.modern_industrialization.machines.init.SingleBlockCraftingMachines;
+import com.google.gson.JsonObject;
+import java.util.ArrayList;
+import java.util.function.Consumer;
+import net.minecraft.resources.ResourceLocation;
+
+public class ExtraMachineConfig {
+
+    public static class CraftingSingleBlock {
+        private final SingleBlockCraftingMachines.Config config;
+
+        public CraftingSingleBlock(SingleBlockCraftingMachines.Config config) {
+            this.config = config;
+        }
+
+        public CraftingSingleBlock steamCustomOverclock(double multiplier, JsonObject object) {
+            var catalysts = new ArrayList<CustomOverclockComponent.Catalyst>();
+
+            for (var entry : object.entrySet()) {
+                catalysts.add(new CustomOverclockComponent.Catalyst(new ResourceLocation(entry.getKey()), entry.getValue().getAsInt()));
+            }
+            config.steamOverclockComponent = new CustomOverclockComponent(multiplier, catalysts);
+            return this;
+        }
+    }
+
+    public static class CraftingMultiBlock {
+        public IOverclockComponent steamOverclockComponent = new GunpowderOverclockComponent();
+        public Consumer<MultiblockMachines.Rei> reiConfig = rei -> {
+        };
+
+        public CraftingMultiBlock steamCustomOverclock(double multiplier, JsonObject object) {
+            var catalysts = new ArrayList<CustomOverclockComponent.Catalyst>();
+
+            for (var entry : object.entrySet()) {
+                catalysts.add(new CustomOverclockComponent.Catalyst(new ResourceLocation(entry.getKey()), entry.getValue().getAsInt()));
+            }
+            steamOverclockComponent = new CustomOverclockComponent(multiplier, catalysts);
+            return this;
+        }
+
+        public CraftingMultiBlock reiExtra(Consumer<MultiblockMachines.Rei> consumer) {
```

Comment:
Maybe it would be wiser to store them in a list, and apply them all, in case people call `reiExtra` multiple times.

Reply:
Right now if you called reiExtra multiple times only the last lambda would be executed. The point of the consumer passing an rei object is that you can chain commands on the object to avoid extra typing (ie builder pattern). If the consumers are stored in a list and applied in order then the last setting still wins.

Is it standard behavior elsewhere in the API to allow calling a builder consumer multiple times? IMO, I'd rather error if they use the method twice and tell them to chain commands and use the builder pattern properly but it's up to you.

Reply:
I agree that one should:tm: use the builder pattern properly, but from the documentation it would be easy to copy/paste multiple `.reiExtra` calls. Sadly not everyone is familiar with programming, and we should try to accommodate for that when we can.

For example, one might copy both `reiExtra(rei => rei.workstations('pyrolyse_oven', 'steam_quarry'))` and `reiExtra(rei => rei.steam(false))`.